pandas/core/reshape/melt.py

```diff
@@ -141,8 +141,7 @@ def melt(
     else:
         mdata[value_name] = frame._values.ravel("F")
     for i, col in enumerate(var_name):
-        # asanyarray will keep the columns as an Index
-        mdata[col] = np.asanyarray(frame.columns._get_level_values(i)).repeat(N)
```

Comment:
excellent!  I've got a comment in my "notes to self branch" on this line:  `# TODO: potential problem with losing EA dtypes?`